rpcs3/rpcs3qt/settings_dialog.cpp

```diff
@@ -599,6 +599,18 @@ settings_dialog::settings_dialog(std::shared_ptr<gui_settings> gui_settings, std
 	});
 	SubscribeTooltip(ui->gbZCULL, tooltips.settings.zcull_operation_mode);
 
+	m_emu_settings->EnhanceComboBox(ui->outputScalingMode, emu_settings_type::OutputScalingMode);
+	SubscribeTooltip(ui->outputScalingMode, tooltips.settings.output_scaling_mode);
+	connect(ui->outputScalingMode, QOverload<int>::of(&QComboBox::currentIndexChanged), this, [r_creator, this](int index)
+	{
+		const bool is_vulkan = (ui->renderBox->currentText() == r_creator->Vulkan.name);
+		const QVariantList var_list = ui->outputScalingMode->itemData(ui->outputScalingMode->currentIndex()).toList();
+		ensure(var_list.size() == 2 && var_list[1].canConvert<int>());
+		const bool fsr_selected = static_cast<output_scaling_mode>(var_list[1].toInt()) == output_scaling_mode::fsr;
+		ui->fsrSharpeningStrength->setEnabled(is_vulkan && fsr_selected);
+		ui->fsrSharpeningStrengthReset->setEnabled(is_vulkan && fsr_selected);
+	});
```

Comment:
I'd rather have one lambda that is triggered whenever we change either settings.

```
const auto enableFsrSettings = [this]()
{
	const bool is_vulkan = (ui->renderBox->currentText() == r_creator->Vulkan.name);
	const QVariantList var_list = ui->outputScalingMode->itemData(ui->outputScalingMode->currentIndex()).toList();
	ensure(var_list.size() == 2 && var_list[1].canConvert<int>());
	const bool fsr_selected = static_cast<output_scaling_mode>(var_list[1].toInt()) == output_scaling_mode::fsr;
	ui->fsrSharpeningStrength->setEnabled(is_vulkan && fsr_selected);
	ui->fsrSharpeningStrengthReset->setEnabled(is_vulkan && fsr_selected);
};
```
`connect(ui->outputScalingMode, QOverload<int>::of(&QComboBox::currentIndexChanged), this, enableFsrSettings);`

Reply:
Will do.

Reply:
I have added an `apply_fsr_specific_options` lambda with the above under the similar `apply_renderer_specific_options` lambda and connected the two in the appropriate spot, removing the duplicate code.